src/Symfony/Bundle/WebProfilerBundle/Tests/DependencyInjection/WebProfilerExtensionTest.php

```diff
@@ -31,6 +31,9 @@ public static function assertSaneContainer(Container $container, $message = '')
     {
         $errors = array();
         foreach ($container->getServiceIds() as $id) {
+            if ('service_container' === $id) { // to be removed in 4.0
```

Comment:
it is non-crashing, but it triggers deprecation warnings everywhere. And there is no way  to avoid this unless you know the ids of all private services (and then, you could as well know the ids of all public services and avoid calling `getServiceIds`)